omigami/recursive_feature_eliminator.py

```diff
@@ -1,37 +1,31 @@
-from typing import List, Dict, Tuple
+from typing import List
 import numpy as np
-from scipy.stats import gmean
-from omigami.feature_evaluator import FeatureEvaluator
-from omigami.models import RecursiveFeatureEliminationResults, SelectedFeatures
-from omigami.utils import average_ranks, normalize_score, get_best_n_features
-from omigami.inner_loop import InnerLoop, InnerLoopResults
+from omigami.data_models import InnerLoopResults
+from omigami.utils import average_ranks, get_best_n_features
 
 
 class RecursiveFeatureEliminator:
     def __init__(
         self,
-        feature_evaluator: FeatureEvaluator,
         dropout_rate: float,
-        robust_minimum: float,
+        n_features: int
     ):
         self.keep_fraction = 1 - dropout_rate
-        self.robust_minimum = robust_minimum
-        self.inner_loop = InnerLoop(feature_evaluator)
-        self._n_features = feature_evaluator.get_n_features()
+        self._n_features = n_features
+        self.features = np.arange(self._n_features)
 
-    def run(self, outer_loop_index):
-        results = {}
-        features = np.arange(self._n_features)
-        while len(features) >= 1:
-            inner_loop_result = self.inner_loop.run(features, outer_loop_index)
-            results[tuple(features)] = inner_loop_result
-            features_to_keep = np.floor(len(features) * self.keep_fraction)
-            features = self._remove_features(inner_loop_result, features_to_keep)
-        avg_scores = self._compute_score_curve(results)
-        best_features = self._select_best_features(results, avg_scores)
-        return RecursiveFeatureEliminationResults(
-            best_feats=best_features, score_vs_feats=avg_scores,
-        )
+    def iter_features(self):
```

Comment:
Do you like this? I find it cumbersome I think having it explicitly out of the class would be better
```python
while features:
    inner_loop_results = ...
    features = rfe.eliminate(inner_loop_results, features)
```
I find it very explicit; however, since also Alan wrote it like this, I you prefer this way sounds good to me

Reply:
I will give another thought on this!